### DISCUSSION TITLE
Developer workshop

### DISCUSSION THREAD

**Discussion by majid-saeed** [TerriaJS/terriajs#7297] [General]
Do you guys have any workshops or some proper development guides?
how can I extend this?
I need popups, I need custom plugins/widgets, map export/print templates, some already present widgets from cesium, attribute tables, I need to integrate routing engine, custom menus, data conversions on export.
where to start?

Using version 8 as a standalone application.

**Comment by ljowen** [comment]
Hi @majid-saeed we could definitely improve the developer onboarding. As a starting point there's content from in person dev workshops we've run in the past: https://github.com/TerriaJS/workshops

Beside that and our docs (https://docs.terria.io/guide/) the source code is the best source of truth.
